components/org.wso2.carbon.deployment.engine/src/test/java/org/wso2/carbon/deployment/engine/deployers/CustomDeployer.java

```diff
@@ -111,7 +131,19 @@ public void undeploy(Object key) throws CarbonDeploymentException {
 
     public String update(Artifact artifact) throws CarbonDeploymentException {
         logger.info("Updating : " + artifact.getName());
-        sample1Updated = true;
+        try (FileInputStream fis = new FileInputStream(artifact.getFile())) {
+            int x = fis.available();
+            byte b[] = new byte[x];
+            fis.read(b);
+            String content = new String(b);
+            if (content.contains("sample1")) {
```

Comment:
Shall we turn-around the equal check for avoiding any NPEs ?
For ex: "something".equals(string)